source/mongodb-handshake/handshake.rst

```diff
@@ -224,6 +224,107 @@ Example::
         - "Oracle JVM EE 9.1.1"
 
 
+--------------------------
+Speculative Authentication
+--------------------------
+
+:since: 4.4
+
+The ``isMaster`` handshake supports a new argument, ``speculativeAuthenticate``,
+provided as a BSON object. Clients specifying this argument to ``isMaster`` will
+speculatively include the first command of an authentication handshake.
+This command may be provided to the server in parallel with any standard request for
+supported authentication mechanisms (i.e. ``saslSupportedMechs``). This would permit
+clients to merge their first message with their ``isMaster`` request, and
+receive the authentication reply with the ``isMaster`` reply.
+
+When the mechanism is ``MONGODB-X509``, `speculativeAuthenticate` has the following
+structure::
+
+    {
+        isMaster: 1,
```

Comment:
I'm not sure it's true that speculativeAuthenticate is exactly the saslStart command document for SCRAM since there is a "db" field. With OP_MSG we would actually send a "$db" field. Perhaps this difference should be explicitly called out?

Reply:
Oh, quite true!  And I missed it because I assumed it was a fork and lift from the auth spec.

So yeah, let's just specify the differences, i.e. the additional fields that are required.

Reply:
I have added text to highlight the differences and added references to the auth spec.